oeps/processes/oep-0056-architectural-process.rst

```diff
@@ -0,0 +1,218 @@
+
+Decisions must document the advice they receive
+-----------------------------------------------
+
+One of the core tenets of the Advisory Process is that it is the decision maker's responsibility to record the advice they received, whether or not they chose to take it, and why.  We recognize that this is a time investment, but believe it will lead to better decisions, better documentation, and an approachable process where stakeholders and experts feel welcomed and encouraged to provide their invaluable insights.
+
+Decentralized decision making
+-----------------------------
+
+We assert that anyone can make an architectural decision, and that the only caveat is that they must consult those who are meaningfully affected by the decision, and those who have expertise in the area the decision is being taken.  The decision and all the advice they receive should be documented in an ADR, co-located with the relevant code following our established ADR guidelines in OEP-19.
+
+Encourage approachable decision making
+--------------------------------------
+
+The process of making a technical decision is not atomic, nor do many parts of it occur while writing a document about the decision.  Many decisions are *made* while writing code as we problem solve and navigate the idiosyncracies of the system.  The goal of our process should be to find a way of sharing and capturing those decisions where they impact others.  In this spirit, we encourage using "low-friction" artifacts - such as GitHub Issues - in those moments where drafting a full ADR document would be prohibitively distracting.  An issue is an easy place to gather quick, initial input, and naturally becomes resolved by a GitHub Pull Request - the ADR - which describes the decision in more detail.
+
+Decisions can have varying impacts and scopes
+---------------------------------------------
+
+The group of impacted individuals for a given decision may vary wildly, and some decisions will necessarily require more input than others.  A developer iterating on new "green field" code may not need to consult *anyone* except their immediate teammates, whereas a team making changes to a core data model may impact stakeholders and systems throughout the platform, meaning that there will necessarily be significant input into the decision.  The volume of input may make the latter decision harder to make.  That the complexity of the decision making process correlates with the impact of the decision indicates we're giving decisions due process.
```

Comment:
This paragraph seems like it might conflict with a Decision Maker checklist.  For example, new component code in an existing repo may have this paragraph's guidance of "only need a team review" but the checklist may say "there are several stakeholders and SMEs for this repo, collect feedback from all of them".  

Should we/How can we differentiate between efforts like refactors and existing feature enhancements VS introduction of new features with little to no dependencies?

Reply:
Hm, I'm not sure those are actually in conflict.  If a repo has stakeholders and SMEs who want to be input, then that changes the impact of the change, and so they should be included.  Examples of this I've seen are around PCI compliance/ecommerce, for instance, where we want to be extra careful and so proposals get extra review.

If new code is going into a less risky repository, then the "green field code" example in the doc may hold true.  

What do you think?  Is there a way I could make this language clearer? 🤔 I may not be getting the spirit of it across.

Reply:
I may be conflating all "work" with decisions requiring ADRs, so apologies upfront for that...

I am considering a case like the Text Editor Overhaul workstream.  We have a repo containing an editor framework and inside it different editor components.  It makes total sense that if someone wants to refactor or enhance the framework, they must seek guidance and approval from the repo owner/SME.  

But, does someone who wants to build a new editor component need to go to the owners of the repo and get approval before they can start?  Additionally, we will have a guide doc outlining how to add a new editor using the framework.  Does the builder still need approval if they are following the guide?

Again, I may be conflating the "build something new" use case with the "ADR-worthy effort" use case.

Reply:
(This is a great line of questioning and I like that we're hashing it through!)

I don't think ADRs are the mechanism by which we decide what work needs to be done, moreso than _how_ it gets done when it affects stakeholders.  

In the "new editor component" situation, what's the impact on the maintainers?  If the decision maker needs to refactor the way the editor works, sure, they'll want to seek input from the SMEs on that refactor(who I'm assuming for the sake of argument are the maintainers).  But if they're adding a net new component that consumers of the text editor can choose whether or not to use, and it's not messing around with the internals, then the impact is small and (presumably) so should the feedback they receive.  

Another way to phrase this is that ADRs are _architectural_ decision records, describing choices we made in the code.  They're not product decision records; the fact that we decided to develop a new component isn't really ADR-worthy unless it has some sort of architectural impact.

Reply:
Yea, I thought I was getting myself mixed up trying to think about how it would play out in practice.  Thanks for walking through it with me.

Reply:
I think it's an excellent callout.  It may be worth adding a little summary of what constitutes an ADR.